guacamole/src/main/webapp/app/navigation/directives/guacUserMenu.js

```diff
@@ -134,6 +134,48 @@ angular.module('navigation').directive('guacUserMenu', [function guacUserMenu()
                 return authenticationService.isAnonymous();
             };
 
+            /**
+             * Requests fullscreen for the Guacamole page.
+             */
+            $scope.fullscreen = function fullscreen() {
+                var elem = document.documentElement;
+                var fullscreenElem = !(!document.fullscreenElement &&
+                    !document.msFullscreenElement &&
+                    !document.mozFullScreenElement &&
+                    !document.webkitFullscreenElement);
+
+                if (!fullscreenElem) {
+                    if (!elem.requestFullscreen) {
+                        elem.requestFullscreen = (elem.mozRequestFullScreen
+                                || elem.webkitRequestFullscreen
+                                || elem.msRequestFullscreen).bind(elem);
+                    }
+
+                    if (elem.requestFullscreen) {
+                        elem.requestFullscreen();
+                    }
```

Comment:
This appears to anticipate that there may not be any `requestFullscreen()` implementation, even browser-specific prefix versions. That's good, but such a case would also cause the previous call to `bind()` to fail, as `elem.mozRequestFullScreen || elem.webkitRequestFullscreen || elem.msRequestFullscreen` would not evaluate to something that has `bind()`.

I'd recommend avoiding both of these issues entirely, adding an anonymous stub function at the end of that chain of `||`, thus ensuring there is always a version of `requestFullscreen()` (even if it does nothing).

Reply:
Ok.

Do you prefer the stub function to really do nothing or is it okay if it logs a warning ?

Reply:
Assuming this is still relevant given what you mentioned regarding hiding the fullscreen option if it can't be used, I have no objection to logging a warning.

Reply:
Given the fact that we will hide the fullscreen option if not supported, I will as well log a warning.